test/unit/repositories/test_item.py

```diff
@@ -77,3 +77,274 @@ def test_create_with_non_existent_system_id(test_helpers, database_mock, item_re
+    catalogue item ID filter
+    """
+    # Mock `find` to return an empty list of item documents
+    test_helpers.mock_find(database_mock.items, [])
+
+    catalogue_item_id = str(ObjectId())
+    retrieved_items = item_repository.list(None, catalogue_item_id)
+
+    database_mock.items.find.assert_called_once_with({"catalogue_item_id": CustomObjectId(catalogue_item_id)})
+    assert retrieved_items == []
+
+
+def test_list_with_invalid_catalogue_item_id(item_repository):
+    """
+    Test getting an item with an invalid catalogue item ID filter
+
+    Verify that the `list` method properly handles the retrieval of items with the provided
+    ID filter
+    """
+    with pytest.raises(InvalidObjectIdError) as exc:
+        item_repository.list(None, "Invalid")
+    assert str(exc.value) == "Invalid ObjectId value 'Invalid'"
+
+
+def test_list_with_both_system_catalogue_item_id_filters(test_helpers, database_mock, item_repository):
+    """
+    Test getting an item with both system and catalogue item id filters.
+
+    Verify that the `list` methof properly handles the retrieval of items with the provided ID filters
+    """
+    item = ItemOut(id=str(ObjectId()), catalogue_item_id=str(ObjectId()), system_id=str(ObjectId()), **FULL_ITEM_INFO)
+
+    # Mock `find` to return a list of item documents
+    test_helpers.mock_find(
+        database_mock.items,
+        [
+            {
+                "_id": CustomObjectId(item.id),
+                "system_id": CustomObjectId(item.system_id),
+                "catalogue_item_id": CustomObjectId(item.catalogue_item_id),
+                **FULL_ITEM_INFO,
+            }
+        ],
+    )
+
+    retrieved_item = item_repository.list(item.system_id, item.catalogue_item_id)
+
+    database_mock.items.find.assert_called_once_with(
+        {"system_id": CustomObjectId(item.system_id), "catalogue_item_id": CustomObjectId(item.catalogue_item_id)}
+    )
+    assert retrieved_item == [item]
+
+
+def test_list_no_matching_result_both_filters(test_helpers, database_mock, item_repository):
+    """
+    Test getting items based on the provided system and catalogue item ID filters when there are no matching results in
+    the database.
+
+    Verify the `list` method properly handles the retrieval of items based on the provided
+    system and catalogue item ID filters
+    """
+    # Mock `find` to return an empty list of item documents
+    test_helpers.mock_find(database_mock.items, [])
+
+    system_id = str(ObjectId())
+    catalogue_item_id = str(ObjectId())
+    retrieved_items = item_repository.list(system_id, catalogue_item_id)
+
+    database_mock.items.find.assert_called_once_with(
+        {"system_id": CustomObjectId(system_id), "catalogue_item_id": CustomObjectId(catalogue_item_id)}
+    )
+    assert retrieved_items == []
+
+
+def test_list_one_filter_no_matching_results(test_helpers, database_mock, item_repository):
```

Comment:
I don't think this test matches the actual behaviour. By supplying both filters, you are essentially doing an `AND` (not an `OR`) query which is what also happens in the query sent to the database. So both filters need to match for the item to be returned otherwise it returns nothing. The only reason it works in this test is because of how the return of the `find` method is mocked which is wrong. Test it manually at the endpoint level and you will see what I mean.